### ISSUE TITLE
SA1515 should not trigger in constructor calls

### ISSUE THREAD

**Issue by TheBrambleShark** [DotNetAnalyzers/StyleCopAnalyzers#3923]
It is not uncommon for me to have large record classes, and often I need to deserialize those either from a DataRow or an array, depending on the use case.

However, commenting out specific lines of code, either for debugging or in cases where the line needs to be preserved is not an uncommon scenario.

```cs
public sealed record class Person
(
    string FirstName,
    string LastName,
    string? SocialSecurityNumber,
    int Age
)
{
    public static Person FromRow(DataRow row)
    {
        return new Person
        (
            FirstName: row.Field<string>(nameof(FirstName))!,
            LastName: row.Field<string>(nameof(LastName))!,
            // SocialSecurityNumber: row.Field<string>(nameof(SocialSecurityNumber)); // SA1515 - Single-line comment should be preceded by blank line.
            Age: row.Field<int>(nameof(Age))
        );
    }
}
```
Commenting out this line of code raises SA1515, but  if you follow its advice by adding an empty row between LastName and SocialSecurityNumber, it instead raises SA1115 (The parameter should begin on the line after the previous parameter).

Current workarounds:
- `#pragma warning disable/enable SA1515` - Clunky, detracts from readability
- Use `////` as comment marker (thought this was a bug, but apparently this is intended behavior).
- Disable SA1515 globally - not ideal because it has valid use cases.

I would expect that SA1515 not be raised in the above example. I feel this is fairly similar to #3766 in nature, as it's a very similar syntax in play.

**Comment by bjornhellander** [comment]
The wording of SA1515 might not be the best in this case, but I see no reason for an exception for this. Single-line comments are intended for documentation and normally requires a preceeding blank line for readability. Temporarily commenting out code can be done with `////` as you have noticed yourself.

**Comment by TheBrambleShark** [comment]
> The wording of SA1515 might not be the best in this case, but I see no reason for an exception for this. Single-line comments are intended for documentation and normally requires a preceeding blank line for readability. Temporarily commenting out code can be done with `////` as you have noticed yourself.

You are correct. I seem to have neglected to close this ticket.
